hosting/opennebula_functions.py

```diff
@@ -0,0 +1,230 @@
+                logger.error("OpenNebulaException error: {0}".format(opennebula_err))
+            except OSError as os_err:
+                messages.add_message(request, messages.ERROR, _("OS error: {0}".format(os_err)))
+                logger.error("OSError : {0}".format(os_err))
+            except ValueError as value_err:
+                messages.add_message(request, messages.ERROR,
+                                     _("Please select an appropriate value for vm template."))
+                logger.error("ValueError : {0}".format(value_err))
+        return redirect('admin:showvms')
+
+    # Delete VM from the pool and DB by using method finalize()
+    def delete_vm(self, request, vmid):
+        vm_id = int(vmid)
+        # get the desired vm from the pool
+        logger.debug("Deleting vm with id {0}".format(vm_id))
+        vm = self.get_vm_by_id(vm_id)
+        if vm is None:
+            messages.add_message(request, messages.ERROR, _("Did not find a vm with id = {0}".format(vm_id)))
+        else:
+            logger.debug("Deleting vm_id = " + str(vm_id) + " state = " + vm.str_state)
+            if vm.str_state == 'PENDING' or vm.str_state == 'POWEROFF' or vm.str_state == 'ACTIVE':
+                vm.delete()
+                messages.add_message(request, messages.SUCCESS,
+                                     _("Deleted from {0} state vm with id = {1}".format(vm.str_state, str(vm_id))))
+            else:
+                vm.finalize()
+                messages.add_message(request, messages.SUCCESS,
+                                     _("Deleted (using finalize()) from {0} state vm with id = {1}".format(vm.str_state,
+                                                                                                           str(vm_id))))
+        return redirect('admin:showvms')
+
+    def stop_vm(self, request, vmid):
+        vm_id = int(vmid)
+        vm = self.get_vm_by_id(vm_id)
+        if vm is None:
+            messages.add_message(request, messages.ERROR, _("Did not find a vm with id = {0}", vm_id))
+        else:
+            vm.stop()
+            messages.add_message(request, messages.SUCCESS, _("Stopped the vm with id = {0}", vm_id))
+        return redirect('admin:showvms')
+
+    def start_vm(self, request, vmid):
+        vm_id = int(vmid)
+        vm = self.get_vm_by_id(vm_id)
+        if vm is None:
+            messages.add_message(request, messages.ERROR, _("Did not find a vm with id = {0}", vm_id))
+        else:
+            vm.resume()
+            messages.add_message(request, messages.SUCCESS, _("Started the vm with id = {0}", vm_id))
+        return redirect('admin:showvms')
+
+    # Retrives virtual machine pool information
+    def get_vm_pool(self):
+        vm_pool = oca.VirtualMachinePool(self.client)
+        vm_pool.info()
+        return vm_pool
+
+    def get_vm_by_id(self, vmid):
+        vm_pool = self.get_vm_pool()
+        return vm_pool.get_by_id(vmid)
```

Comment:
If no vm is found this method raises a WrongIdError, can we make sure to catch it? You could replace the `if vm is None` since it's useless on start_vm and stop_vm